docs/docs/topics/developer-guide.mdx

```diff
@@ -0,0 +1,133 @@
+
+#### Typing & Protocols (Python)
+
+Infrahub supports Python Typing and Protocols to improve the developer experience by providing static analysis and detecting issues early.
+
+:::info What are Protocols?
+
+Protocols are a way to define structural subtyping in Python, allowing you to specify expected behavior without requiring inheritance. Infrahub leverages this to enforce typing for flexible schemas.
+
+:::
+
+##### Generating Protocols
+
+You can generate Protocols for your schema with the `infrahubctl protocols` command. The schema can be provided locally or pulled from a running Infrahub instance.
+
+```shell
+infrahubctl protocols --schema my-schema.yaml --output protocols.py
+```
+
+> If you don’t have a Python module, save the protocols in a protocols.py file in the same directory as your script.
+
+##### Using Protocols in Code
+
+The python SDK natively support Protocol in place of `kind` for most functions that interact with objects (`all`, `filters`, `get`, `create`)
+
+```python
+# With Protocols
+from infrahub_sdk.protocols import BuiltinTag
+
+tags = await client.all(kind=BuiltinTag)
+for tag in tags:
+    print(tag.name.value)  # Type-safe access to the 'name' attribute
+```
+
+Same code without Protocols
+
+```python
+# Without Protocols
+tags = await client.all(kind="BuiltinTag")
+for tag in tags:
+    print(tag.name.value)  # May raise type errors as attributes are not checked
+```
+
+:::info Protocol for Infrahub Schema
+
+Protocols for all objects natively provided by Infrahub are available within the SDK in `infrahub_sdk.protocols`
+
+:::
+
+##### Learn More about Python Protocols
+
+- [Python Protocols: Leveraging Structural Subtyping - Real Python](https://realpython.com/python-protocol/)
+- [What are "Protocols" In Python? - Youtube](https://www.youtube.com/watch?v=2jN11lyKvfA)
+
+#### Git integration
+
+Infrahub integrates seamlessly with Git, allowing you to use your current branch name as the default branch. To enable this feature, set the environment variable:
+
+```shell
+export INFRAHUB_DEFAULT_BRANCH_FROM_GIT=true
+```
+
+### Testing
+
+Testing is crucial for maintaining high-quality code. Infrahub provides a testing framework based on Pytest to simplify unit and integration tests.
+
+Refer to the [Testing Framework Documentation](./resources-testing-framework.mdx) for details.
+
+:::warning Under Construction
+
+This section is still under development. Contact the team via Discord for additional resources.
+
+:::
+
+## Logging (Python)
+
+You can use Python’s standard logging library to gain insights into the execution of your code. Infrahub captures logs from the `infrahub.tasks` logger by default.
+
+The logger `infrahub.tasks` is configure by default and it's possible to configure additional loggers using the settings (see below)
+
+```python
+import logging
+
+log = logging.getLogger("infrahub.tasks")
```

Comment:
Just a thought for maybe our Python and generators, should we be instantiating a logging object to be used to bubble up logs from these rather than leaving it to users to get the `infrahub.tasks` logger?

Reply:
Agreed, this can be done in the SDK directly so not a blocker for this PR